chain-vote/src/lib.rs

```diff
@@ -1,27 +1,28 @@
+// These two need to be here because the points of sec2 backend do not implement Copy
+#![allow(clippy::clone_on_copy)]
+#![allow(clippy::op_ref)]
```

Comment:
This should be redundant here now?
I don't think the blanket clippy overrides for the whole library is a good idea, too. It's better to keep them close in the modules where the overrides are needed (and maybe think twice whether they should be removed by redesign, as long as we're changing the API anyway)

Reply:
Why redundant? You mean that not both are needed? As far as I can tell, the first (`clone_on_copy`) is required for when you clone a structure that implements copy (and is therefore not required). The second, is when you unnecessarily use a reference. 

I also agree that it's better to no override the whole library, but in this case, it seems required. The problem is that the two ec backends differ in the `GroupElement` and `Scalar` structure. `eccoxide` does not implement the copy trait, so we need to clone everything, and use references rather than values. On the other hand, `curve25519_dalek` does. Given that the latter is our default feature, it will complain. And the reason to have it in the library is because this flag is needed in almost every operation where `GroupElements` or `Scalar`. However, if it is preferable, and can put it in every module individually 👍

Reply:
OK then, leave it.
Would it be annoying to not implement `Copy` instead?

Reply:
very good point 👍  much cleaner now.